developer/src/kmc-ldml/src/compiler/keys.ts

```diff
@@ -446,7 +446,7 @@ export class KeysCompiler extends SectionCompiler {
 
     if (layer.row.length > keymap.length) {
       this.callbacks.reportMessage(
-        LdmlCompilerMessages.Error_HardwareLayerHasTooManyRows()
+        LdmlCompilerMessages.Error_HardwareLayerHasTooManyRows(layer)
```

Comment:
@mcdurdin and all, this is one of the main things i want to review - the ergonomics on the call site for passing in this info.  This way there's no extra helper function invocation